services/azure-devops/azure-devops-coverage.tester.js

```diff
@@ -0,0 +1,215 @@
+'use strict'
+
+const Joi = require('joi')
+const { withRegex } = require('../test-validators')
+const t = require('../create-service-tester')()
+module.exports = t
+
+const service = 'azure-devops'
+const noun = 'coverage'
+const org = 'swellaby'
+const project = 'opensource'
+const linuxDefinitionId = 21
+const macDefinitionId = 26
+const windowsDefinitionId = 24
+const nonExistentDefinitionId = 234421
+const buildId = 946
+const uriPrefix = `/${service}/${noun}/${org}/${project}`
+const azureDevOpsApiBaseUri = `https://dev.azure.com/${org}/${project}/_apis`
+const mockBadgeUriPath = `${uriPrefix}/${macDefinitionId}.json`
+const mockLatestBuildApiUriPath = `/build/builds?definitions=${macDefinitionId}&$top=1&api-version=5.0-preview.4`
+const mockCodeCoverageApiUriPath = `/test/codecoverage?buildId=${buildId}&api-version=5.0-preview.1`
+const percentageRegex = /^[1-9][0-9]?%|^100%|^0%$/
+const latestBuildResponse = {
+  count: 1,
+  value: [{ id: buildId }],
+}
+const firstLinesCovStat = {
+  label: 'Lines',
+  total: 23,
+  covered: 19,
+}
+
+const branchCovStat = {
+  label: 'Branches',
+  total: 11,
+  covered: 7,
+}
+
+const secondLinesCovStat = {
+  label: 'Lines',
+  total: 41,
+  covered: 35,
+}
+
+const firstLinesCoverage = `${(
+  (firstLinesCovStat.covered / firstLinesCovStat.total) *
+  100
+).toFixed(0)}%`
+const multiLinesTotal = firstLinesCovStat.total + secondLinesCovStat.total
+const multiLinesCovered = firstLinesCovStat.covered + secondLinesCovStat.covered
+const multiLinesCoverage = `${(
+  (multiLinesCovered / multiLinesTotal) *
+  100
+).toFixed(0)}%`
```

Comment:
Would it be possible to simplify the above and make it more direct by using literal or explicit fixtures? While it's nice to use symbolic names to DRY up assertions, as I'm not familiar with this API, it's difficult to follow what's happening.

Reply:
Yup, will do. My initial motivation was as much about trying to avoid doing the math myself as anything else but making this more readable as you suggest will be better

Reply:
I did some condensing here but was wondering if you're asking instead to use literals in each test? Happy to make that change if so. 

I definitely learned what the Tutorial means when it says that nock is "fussy" so once I got those paths just right I wanted to ensure it would be easy to update them as the paths/query params/etc. change as the Previews of the the underlying APIs progress